src/zenml/zen_stores/models/component_wrapper.py

```diff
@@ -0,0 +1,96 @@
+#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+
+import base64
+import json
+from uuid import UUID
+
+import yaml
+from pydantic import BaseModel
+
+from zenml.enums import StackComponentType
+from zenml.logger import get_logger
+from zenml.stack import StackComponent
+
+logger = get_logger(__name__)
+
+
+class ComponentWrapper(BaseModel):
+    """Serializable Configuration of a StackComponent"""
+
+    type: StackComponentType
+    flavor: str
+    name: str
+    uuid: UUID
+    config: bytes  # b64 encoded yaml config
+
+    @classmethod
+    def from_component(cls, component: StackComponent) -> "ComponentWrapper":
+        """Creates a ComponentWrapper from an actual instance of a Stack
+        Component.
+
+        Args:
+            component: the instance of a StackComponent
+        """
+        return cls(
+            type=component.TYPE,
+            flavor=component.FLAVOR,
+            name=component.name,
+            uuid=component.uuid,
+            config=base64.b64encode(
+                yaml.dump(json.loads(component.json())).encode()
+            ),
+        )
+
+    def to_component(self) -> StackComponent:
+        """Converts the ComponentWrapper into an actual instance of a Stack
+        Component."""
+        from zenml.repository import Repository
+        from zenml.stack.registry import flavor_registry
+
+        zenml_flavors = flavor_registry.get_flavors_by_type(
+            component_type=self.type
```

Comment:
I would like to echo @schustmi 's concerns about this. The FlavorRegistry is basically a thin in-memory read-only "layer" or "shim" that is applied on top of whatever store is the active one. Because of this aspect, there is now a need to implement additional logic to enforce this convention and to make sure that all the calls to "get_flavor" and "register_flavor" first go through the FlavorRegistry and then to the ZenStore.

Ideally, this logic should be located under one class that acts as the go-to object for all flavor related operations, that exposes a public API that is consumed by all other areas of the code that need to access flavors (e.g. the CLI, the REST API, ZenML custom user code). For things like stacks and stack components, this one class used to be the Repository, that had the role of acting as the central management point for stacks/components. I would recommend that we continue to use Repository to also enforce this new convention about flavors.

This basically means that Repository should expose `get_flavor`, `register_flavor` etc. as flavor management methods and that everywhere else in the code, we use `Repository().get_flavor(...)` and `Repository().register_flavor(...)` as the official API for flavor management. The Repository instead is responsible for coordinating between FlavorRegistry and ZenStore to implement these methods. This separation of concerns hides the back-end complexity away from users.

Reply:
Tried to reflect on my answer in the comment below. Let me know if you have further questions.